flight/Modules/Sensors/sensors.c

```diff
@@ -209,8 +213,28 @@ static void SensorsTask(void *parameters)
 		}
 
 		queue = PIOS_SENSORS_GetQueue(PIOS_SENSOR_BARO);
-		if (queue != NULL && PIOS_Queue_Receive(queue, &baro, 0) != false) {
-			update_baro(&baro);
+		if (queue != NULL) {
+			if (PIOS_Queue_Receive(queue, &baro, 0) != false) {
+				// we can use the timeval because it contains the current time stamp (PIOS_DELAY_GetRaw())
+				last_baro_update_time = timeval;
+				update_baro(&baro);
+				// if we were in baro error state clear it.
+				// To avoid sending the clear signal all the time
+				// "in_baro_error_state" is used.
+				if (in_baro_alarm_state) {
```

Comment:
No need for this condition or tracking the state. Just call AlarmsClear directly. The library takes care of checking the previous condition https://github.com/TauLabs/TauLabs/blob/next/flight/Libraries/alarms.c#L74